frontend/lib/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -101,12 +102,21 @@ export interface Props {
  * Splits the string on non-alphanumeric characters, and joins with a dash.
  */
 export function createAnchorFromText(text: string | null): string {
-  const newAnchor = text
-    ?.toLowerCase()
-    .split(/[^A-Za-z0-9]/)
-    .filter(Boolean)
-    .join("-")
-  return newAnchor || ""
+  let newAnchor = ""
+  // Check if the text is valid ASCII characters - necessary for fully functional anchors (issue #5291)
+  const isASCII = text && /^[\x00-\x7F]*$/.test(text)
+
+  if (isASCII) {
+    newAnchor = text
+      ?.toLowerCase()
+      .split(/[^\p{L}\p{N}]+/gu) // split on non-alphanumeric characters
+      .filter(Boolean)
```

Comment:
Might be helpful to comment what the filter line is doing (if you know it), while you're making changes to this part. Since it isn't obvious to me, it probably isn't obvious to others either.